pkg/zpdb/config_test.go

```diff
@@ -0,0 +1,282 @@
+
+	pdbCfg.maxUnavailablePercentage = 1
+	require.Equal(t, 1, pdbCfg.MaxUnavailablePods(sts))
+
+	pdbCfg.maxUnavailablePercentage = 75
+	require.Equal(t, 2, pdbCfg.MaxUnavailablePods(sts))
+}
+
+// TestPodPartitionZoneMatch validates the regular expression parsing of a Pod name to return a logical partition name
+func TestPodPartitionZoneMatch(t *testing.T) {
+	pdbCfg := Config{
+		podNamePartition:           regexp.MustCompile(`^[a-z\-]+-(zone-[a-z]-[0-9]+)$`),
+		podNamePartitionRegexGroup: 1,
+	}
+
+	// test successful matches
+	for _, name := range []string{"ingester-zone-a-0", "test-app-zone-a-0"} {
+		pod := newPodCfgTest(name)
+		p, err := pdbCfg.PodPartition(pod)
+		require.NoError(t, err)
+		require.Equal(t, "zone-a-0", p)
+	}
+
+	// test no match
+	for _, name := range []string{"", "ingester-zone-a", "test-app-1"} {
+		pod := newPodCfgTest(name)
+		_, err := pdbCfg.PodPartition(pod)
+		require.NotNil(t, err)
+	}
+}
+
+// TestPodPartitionZoneMatch validates the regular expression parsing of a Pod name to return a logical partition name
+func TestPodPartitionZoneMatchWithGrouping(t *testing.T) {
+	pdbCfg := Config{
+		podNamePartition:           regexp.MustCompile(`^ingester(-foo)?-zone-[a-z]-([0-9]+)$`),
+		podNamePartitionRegexGroup: 2,
+	}
+
+	// test successful matches
+	for _, name := range []string{"ingester-foo-zone-a-0", "ingester-zone-a-0"} {
+		pod := newPodCfgTest(name)
+		p, err := pdbCfg.PodPartition(pod)
+		require.NoError(t, err)
+		require.Equal(t, "0", p)
+	}
+}
+
+func TestBadRegex(t *testing.T) {
+	raw := map[string]interface{}{}
+	raw[fieldPodNamePartitionRegex] = "(a bad regex["
+
+	_, _, err := valueAsRegex(raw, fieldPodNamePartitionRegex, fieldPodNameRegexGroup)
+	require.ErrorContains(t, err, "error parsing regexp: missing closing ]: `[$`")
+
+	raw[fieldPodNamePartitionRegex] = "ingester-zone-[a-z]-[0-9]+"
+	_, _, err = valueAsRegex(raw, fieldPodNamePartitionRegex, fieldPodNameRegexGroup)
+	require.ErrorContains(t, err, "regular expression requires at least one subexpression")
+
+	raw[fieldPodNamePartitionRegex] = "ingester-zone-([a-z])-([0-9]+)"
+	_, _, err = valueAsRegex(raw, fieldPodNamePartitionRegex, fieldPodNameRegexGroup)
+	require.ErrorContains(t, err, "regular expression has multiple subexpressions and requires podNameRegexGroup to be set")
+
+	raw[fieldPodNameRegexGroup] = int64(10)
+	raw[fieldPodNamePartitionRegex] = "ingester-zone-([a-z])-([0-9]+),$10"
+	_, _, err = valueAsRegex(raw, fieldPodNamePartitionRegex, fieldPodNameRegexGroup)
+	fmt.Printf("%s\n", err.Error())
+	require.ErrorContains(t, err, "regular expression subexpression index out of range")
+}
+
+func TestRegexGroupingDefault(t *testing.T) {
+	raw := map[string]interface{}{
+		fieldPodNamePartitionRegex: "ingester-zone-[a-z]-([0-9]+)",
+	}
+	regex, group, err := valueAsRegex(raw, fieldPodNamePartitionRegex, fieldPodNameRegexGroup)
+	require.NotNil(t, regex)
+	require.Equal(t, 1, group)
+	require.Nil(t, err)
```

Comment:
[nit] The `require.Error` and `require.NoError` assertions produce slightly nicer error messages:

```suggestion
	require.NoError(t, err)
```

(same comment applies in many other places)